src/puppeteer/puppeteer-plugin.ts

```diff
@@ -6,23 +6,35 @@ import { PuppeteerController } from './puppeteer-controller';
 
 const PROXY_SERVER_ARG = '--proxy-server=';
 
-/**
- * Puppeteer
- */
 export class PuppeteerPlugin extends BrowserPlugin<typeof Puppeteer> {
     protected async _launch(launchContext: LaunchContext<typeof Puppeteer>): Promise<Puppeteer.Browser> {
         const {
             launchOptions,
             anonymizedProxyUrl,
             userDataDir,
+            useIncognitoPages,
         } = launchContext;
 
         const finalLaunchOptions = {
             ...launchOptions,
             userDataDir: launchOptions?.userDataDir ?? userDataDir,
         };
 
-        const browser = await this.library.launch(finalLaunchOptions);
+        let browser = await this.library.launch(finalLaunchOptions);
+        if (useIncognitoPages) {
+            browser = new Proxy(browser, {
+                get: (target, property: keyof typeof browser) => {
+                    if (property === 'newPage') {
+                        return (async (...args) => {
+                            const incognitoContext = await browser.createIncognitoBrowserContext();
+                            return incognitoContext.newPage(...args);
+                        }) as typeof browser.newPage;
+                    }
+
+                    return target[property];
```

Comment:
We need to make sure this does not clash with the `useIncognitoPages` implementation in the controller. I guess that one should be removed if we add this.

Reply:
Fixed.